backend/eventloop/application_event_loop/application_event_runner.go

```diff
@@ -97,6 +98,10 @@ func applicationEventLoopRunner(inputChannel chan *eventlooptypes.EventLoopEvent
 
 		ctx, cancel := context.WithCancel(outerContext)
 
+		if newEvent.Request.ClusterName != "" && !sharedutil.IsKCPVirtualWorkspaceDisabled() {
```

Comment:
Looks like we're repeating this snippet of code a few times, is it worth it to create a function for it in shared util, something like:
```go
	func addKCPClusterToContext(ctx context.Context, string clusterName) context.Context {
		if clusterName != "" && !sharedutil.IsKCPVirtualWorkspaceDisabled() {
			ctx = logicalcluster.WithCluster(ctx, logicalcluster.New(newEvent.Request.clusterName))
			return ctx
		} else {
			return ctx
		}
	}
```
Then calling it like
```go
	ctx = addKCPClusterToContext(ctx, newEvent.Request.ClusterName)
```

Reply:
Updated